chain/txValidator.go

```diff
@@ -145,12 +145,16 @@ func CheckTransactionPayload(txn *transaction.Transaction, height uint32) error
 			return errors.New("Register name transaction is not supported yet")
 		}
 		pld := payload.(*pb.RegisterName)
-		match, err := regexp.MatchString("(^[A-Za-z][A-Za-z0-9-_.+]{2,254}$)", pld.Name)
+		regexPattern := "(^[A-Za-z][A-Za-z0-9+]{5,254})\\.nkn$"
```

Comment:
1. I think it's better to restrict name to < 64 char length to avoid being confused with pubkey
2. Why we enforce `.nkn` suffix? That should be part of the application or parse logic, but not in the database level

Reply:
Also this regexPattern could be represented by a HeightDependent